kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/json/JsonRecordWriterProvider.java

```diff
@@ -76,6 +77,18 @@ public RecordWriter getRecordWriter(final S3SinkConnectorConfig conf, final Stri
         public void write(SinkRecord record) {
           log.trace("Sink record with view {}: {}", recordView, record);
           try {
+            // headers need to be enveloped for json format
```

Comment:
does this envelope still work for avro and other formats?

Reply:
Headers don't require the additional enveloped format when using the other writer classes. It's a unique requirement for headers with the Json writer.

Reply:
right, so we are enveloping all of the headers regardless of the format now, is that correct? while before we did not envelop avro, which means this fix isn't backwards incompatible or am i missing something?

Reply:
This change is only for the `JsonRecordWriterProvider`. `Avro` and other formats are not affected by this PR.

Reply:
I am blind and I apologize

Reply:
all good, complex feature :)